packages/api-db-mongodb/src/db/comment.ts

```diff
@@ -0,0 +1,184 @@
+import {
+  DBCommentAdapter,
+  CreateCommentArgs,
+  Comment,
+  GetCommentsArgs,
+  ConnectionResult,
+  CommentState,
+  LimitType,
+  InputCursorType,
+  CommentSort,
+  SortOrder,
+  PublicComment,
+  CommentAuthorType
+} from '@wepublish/api'
+
+import {Collection, Db, FilterQuery, MongoCountPreferences} from 'mongodb'
+import {Cursor} from './cursor'
+import {MaxResultsPerPage} from './defaults'
+
+import {CollectionName, DBComment} from './schema'
+
+function commentSortFieldForSort(sort: CommentSort) {
+  switch (sort) {
+    case CommentSort.CreatedAt:
+      return 'createdAt'
+
+    case CommentSort.ModifiedAt:
+      return 'modifiedAt'
+  }
+}
+
+function commentDateForSort(comment: DBComment, sort: CommentSort): Date {
+  switch (sort) {
+    case CommentSort.CreatedAt:
+      return comment.createdAt
+
+    case CommentSort.ModifiedAt:
+      return comment.modifiedAt
+  }
+}
+
+export class MongoDBCommentAdapter implements DBCommentAdapter {
+  private comments: Collection<DBComment>
+  private locale: string
+
+  constructor(db: Db, locale: string) {
+    this.comments = db.collection(CollectionName.Comments)
+    this.locale = locale
+  }
+
+  async createComment({input}: CreateCommentArgs): Promise<PublicComment> {
+    const {text, ...data} = input
+    const {ops} = await this.comments.insertOne({
+      ...data,
+      state: CommentState.PendingApproval,
+      authorType: CommentAuthorType.Author,
+      revisions: [
+        {
+          text,
+          createdAt: new Date()
+        }
+      ],
+      createdAt: new Date(),
+      modifiedAt: new Date()
+    })
+
+    const {_id: id, revisions, ...comment} = ops[0]
+
+    return {
+      ...comment,
+      id,
+      text: revisions[revisions.length - 1].text
```

Comment:
might need a sort by `createdAt`

Reply:
no need in `addPublicComment` method I guess, but it's needed in the queries yes